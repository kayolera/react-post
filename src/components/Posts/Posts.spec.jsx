const { render, screen } = require('@testing-library/react');
import { Posts } from '.';

const props = {
  posts: [
    {
      id: 1,
      cover: 'cover',
      body: 'Body',
      title: 'Title',
    },
    {
      id: 2,
      cover: 'cover 2',
      body: 'Body 2',
      title: 'Title 2',
    },
    {
      id: 3,
      cover: 'cover 3',
      body: 'Body 3',
      title: 'Title 3',
    },
  ],
};

describe('<Posts />', () => {
  it('should render posts', () => {
    render(<Posts {...props} />);
    expect(screen.getAllByRole('heading', { name: /Title/i })).toHaveLength(3);
  });

  it('should not render posts', () => {
    render(<Posts />);
    expect(screen.queryAllByRole('heading', { name: /Title/i })).toHaveLength(0);
  });

  it('should match snapshot', () => {
    const { container } = render(<Posts {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
