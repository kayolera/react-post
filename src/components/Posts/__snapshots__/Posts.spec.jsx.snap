// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="Title"
      src="cover"
    />
    <div
      class="post-content"
    >
      <h2>
        Title
      </h2>
      <p>
        Body
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Title 2"
      src="cover 2"
    />
    <div
      class="post-content"
    >
      <h2>
        Title 2
      </h2>
      <p>
        Body 2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Title 3"
      src="cover 3"
    />
    <div
      class="post-content"
    >
      <h2>
        Title 3
      </h2>
      <p>
        Body 3
      </p>
    </div>
  </div>
</div>
`;
